backends/utils/history_server.go

```diff
@@ -0,0 +1,617 @@
+	LogsFolderPath                 string
+	LogsContainer                  string
+	MaxBytesInNginxRequest         int
+	HistoryFileDurationSecondSpans int64
+	HistoryFileNum                 int
+}
+
+func NewHistoryServer(logger logger.Logger, cfg *frames.Config) *HistoryServer {
+	mon := HistoryServer{
+		logger:   logger,
+		requests: make(chan HistoryEntry, 100),
+		isActive: false,
+		config:   cfg}
+
+	mon.initDefaults()
+
+	mon.createDefaultV3ioClient()
+	if mon.isActive {
+		mon.Start()
+		mon.StartEvictionTask()
+	}
+	return &mon
+}
+
+func (m *HistoryServer) initDefaults() {
+	m.WriteMonitoringLogsTimeout = defaultWriteMonitoringLogsTimeout
+	if m.config.WriteMonitoringLogsTimeoutSeconds != 0 {
+		m.WriteMonitoringLogsTimeout = time.Duration(m.config.WriteMonitoringLogsTimeoutSeconds) * time.Second
+	}
+
+	m.PendingLogsBatchSize = defaultPendingLogsBatchSize
+	if m.config.PendingLogsBatchSize != 0 {
+		m.PendingLogsBatchSize = m.config.PendingLogsBatchSize
+	}
+
+	m.LogsFolderPath = defaultLogsFolderPath
+	if m.config.LogsFolderPath != "" {
+		m.LogsFolderPath = m.config.LogsFolderPath
+	}
+
+	m.LogsContainer = defaultLogsContainer
+	if m.config.LogsContainer != "" {
+		m.LogsContainer = m.config.LogsContainer
+	}
+
+	m.MaxBytesInNginxRequest = defaultMaxBytesInNginxRequest
+	if m.config.MaxBytesInNginxRequest != 0 {
+		m.MaxBytesInNginxRequest = m.config.MaxBytesInNginxRequest
+	}
+
+	m.HistoryFileDurationSecondSpans = defaultHistoryFileDurationInSeconds
+	if m.config.HistoryFileDurationHourSpans != 0 {
+		m.HistoryFileDurationSecondSpans = m.config.HistoryFileDurationHourSpans * 3600
+	}
+
+	m.HistoryFileNum = defaultHistoryFileNum
+	if m.config.HistoryFileNum != 0 {
+		m.HistoryFileNum = m.config.HistoryFileNum
+	}
+}
+
+func (m *HistoryServer) getCurrentLogFileName() string {
+	// Round current time to the nearest log time-bucket
+	currentTimeInSeconds := time.Now().Unix()
+	currentTimeBucket := m.HistoryFileDurationSecondSpans * (currentTimeInSeconds / m.HistoryFileDurationSecondSpans)
+	return fmt.Sprintf("%v/%v.json", m.LogsFolderPath, currentTimeBucket)
+}
+
+func (m *HistoryServer) getLogFileNameByTime(timeBucket int64) string {
+	return fmt.Sprintf("%v/%v.json", m.LogsFolderPath, timeBucket)
+}
+
+func (m *HistoryServer) Start() {
+	go func() {
+		var pendingLogs []HistoryEntry
+
+		for {
+			select {
+			case entry := <-m.requests:
+				pendingLogs = append(pendingLogs, entry)
+				m.logs = append(m.logs, entry)
+
+				if len(pendingLogs) == m.PendingLogsBatchSize {
+					m.writeMonitoringBatch(pendingLogs)
+					pendingLogs = pendingLogs[:0]
+				}
+			case <-time.After(m.WriteMonitoringLogsTimeout):
+				if len(pendingLogs) > 0 {
+					m.writeMonitoringBatch(pendingLogs)
+					pendingLogs = pendingLogs[:0]
+				}
+			}
+		}
+	}()
+}
+
+func (m *HistoryServer) createDefaultV3ioClient() {
+
+	session := &frames.Session{}
+
+	token := "99b630dd-bcec-4d3f-a77e-31a3761cef28" // os.Getenv("V3IO_ACCESS_KEY")
```

Comment:
🙄  in the meantime for things to pass i left it. because until platform changes will be pushed no one sets that environment variable.
But ill change it in the PR anyways